src/native/eventpipe/ep-session.c

```diff
@@ -521,6 +523,19 @@ ep_session_write_event (
 
 	// Filter events specific to "this" session based on precomputed flag on provider/events.
 	if (ep_event_is_enabled_by_mask (ep_event, ep_session_get_mask (session))) {
+		if (ep_session_get_rundown_enabled (session) && (ep_session_get_rundown_thread_id (session) != ep_thread_get_os_thread_id (ep_thread_get ()))) {
+			EP_ASSERT (ep_session_get_rundown_thread_id (session) != 0);
+			return false;
+		}
+
+		if (ep_session_get_rundown_enabled (session) && (ep_session_get_rundown_thread_id (session) == ep_thread_get_os_thread_id (ep_thread_get ()))) {
```

Comment:
Is 531-537 just for debugging? If so can we put an `#ifdef` around it? If not failfast seems like an overly aggresive response to logging something we didn't expect to log.

Reply:
I see you guys were discussing having this fail in CI. That seems fine to me, but I assume the code comes back out before the PR is merged?

Reply:
Right, this code will be backed out. It's just to verify that we aren't getting rundown events from the public provider so we can be confident in removing the public provider rather than doing the thread filtering.

My plan is to run a diagnostc repo test pass with these changes and then call it good and change this PR to remove the public provider rather than filter by thread

Reply:
Sounds good, that satisfies all my concerns around the failfast stuff :)